File: packages/python/src/agent_capture/_internal/runtime.py
"""Process-wide singletons for the running capture engine.

The :func:`agent_capture.configure` entry point registers a default
:class:`SpanBuilder` here. The :func:`agent_capture.instrumentation.decorator.traced`
helper reads from it. Tests can either call :func:`set_default_builder`
explicitly or pass an explicit builder to ``traced(..., builder=...)``.
"""

from __future__ import annotations

from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from agent_capture.span.builder import SpanBuilder

_default_builder: SpanBuilder | None = None


def set_default_builder(builder: SpanBuilder | None) -> None:
    """Register the process-wide default builder. Pass ``None`` to clear."""
    global _default_builder
    _default_builder = builder


def default_builder() -> SpanBuilder | None:
    """Return the process-wide default builder, or ``None`` if unconfigured."""
    return _default_builder

File: packages/typescript/src/internal/runtime.ts
/**
 * Process-wide singletons. Mirrors agent_capture._internal.runtime.
 *
 * `configure()` registers the default SpanBuilder here so traced() and
 * tracedScope() can find it without an explicit argument.
 */

import type { SpanBuilder } from "../span/builder.js";

let _defaultBuilder: SpanBuilder | null = null;

export function setDefaultBuilder(builder: SpanBuilder | null): void {
  _defaultBuilder = builder;
}

export function defaultBuilder(): SpanBuilder | null {
  return _defaultBuilder;
}
